### PR TITLE
CFE-3178: Added support for CIFS and PanFS file systems to storage promise type (3.12)

### PR DESCRIPTION
**Pull request by olehermanse**
Original patch submitted by GitHub user @favoritelotus:

https://github.com/cfengine/core/pull/3362

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2995/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2995/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2995/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2997/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2997/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2997/

**Comment by olehermanse** [comment]
@vpodzime Please review, I will forward port to master once merged :)

**Comment by olehermanse** [comment]
@favoritelotus I've made quite some changes to your patches, could you test if they still work for you?

Here is the current diff of this PR (apply it to 3.12 sources before building):

https://patch-diff.githubusercontent.com/raw/cfengine/core/pull/3893.diff

(Or check out this Pull request branch using git, and build from there, whichever is easier).

Since this is a new feature, it will likely be introduced in 3.15.0, and not in a patch release of 3.12, but getting your testing on the 3.12.x version in this PR is still valuable.

Also, @nickanderson maybe you can test that CIFS/SMB and NFS works with storage promises? Enterprise packages are here:

http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2997/

File: cf-agent/nfs.c [hunk 1]
```diff
@@ -126,14 +128,58 @@ static const char *const VMOUNTOPTS[] =
     [PLATFORM_CONTEXT_ANDROID] = "defaults",         /* android */
 };
 
+static void GetHostAndSource(const char *buf, char *host, char *source)
+/* Extracts from buffer host & source of nfs, cifs and panfs network systems */
+{
```
Comment (vpodzime) [inline cf-agent/nfs.c]: I'd add `assert(StringStartsWith(buf, "panfs://") || StringStartsWith(buf, "//") || strchr(buf, ":"))` here.

Comment (olehermanse) [reply cf-agent/nfs.c]: Alright. I'll leave this PR as is and do that in the master PR.

**Comment by olehermanse** [comment]
Master PR: https://github.com/cfengine/core/pull/3903

File: cf-agent/nfs.c (hunk 1 above)
Comment (olehermanse) [reply cf-agent/nfs.c]: Added in master PR: https://github.com/cfengine/core/pull/3903

**Comment by olehermanse** [comment]
@favoritelotus I will close this PR now since we won't merge it (to 3.12.x branch). Feel free to still comment here if / when you have time to test this.

JIRA ticket: https://tracker.mender.io/browse/CFE-3178

Master PR: https://github.com/cfengine/core/pull/3903

**Pull request closed without merging**